### PR TITLE
Identical twoWayKey

### PR DESCRIPTION
**Pull request by fogelito**
This issue was asked during office hours, Trying to create two relations on the same collection.
So now officially we can...

### CODE REVIEW

File: src/Database/Database.php [hunk 1]
```diff
@@ -1474,6 +1474,13 @@ public function createRelationship(
             if (\strtolower($attribute->getId()) === \strtolower($id)) {
                 throw new DuplicateException('Attribute already exists');
             }
+
+            if ($attribute->getAttribute('type') === self::VAR_RELATIONSHIP
+                && \strtolower($attribute->getAttribute('options')['twoWayKey']) === \strtolower($twoWayKey)
+                && $attribute->getAttribute('options')['relatedCollection'] === $relatedCollection->getId()
+            ) {
+                throw new DuplicateException('TwoWayKey already exists');
```
Comment (abnegate) [inline src/Database/Database.php:1482]: Let's use `Related attribute already exists` for the message here

File: tests/Database/Base.php [hunk 2]
```diff
@@ -4733,6 +4733,81 @@ public function testOneToOneTwoWayRelationship(): void
...
+            $this->expectNotToPerformAssertions();
+            return;
+        }
+
+        static::getDatabase()->createCollection('parent');
+        static::getDatabase()->createCollection('son');
+
+        static::getDatabase()->createRelationship(
+            collection: 'parent',
+            relatedCollection: 'son',
```
Comment (abnegate) [inline tests/Database/Base.php:4748]: Let's use `child` instead of `son` to stay gender neutral

File: src/Database/Database.php (hunk 1 above)
Comment (fogelito) [reply src/Database/Database.php:1482]: Done!

File: tests/Database/Base.php (hunk 2 above)
Comment (fogelito) [reply tests/Database/Base.php]: Done! 👍

File: src/Database/Database.php [hunk 3]
```diff
@@ -1663,7 +1670,7 @@ public function updateRelationship(
                 !\is_null($newTwoWayKey)
                 && \in_array($newTwoWayKey, \array_map(fn ($attribute) => $attribute['key'], $relatedAttributes))
             ) {
-                throw new DuplicateException('Attribute already exists');
+                throw new DuplicateException('Related attribute already exists');
```
Comment (fogelito) [inline src/Database/Database.php:1673]: @abnegate please double check me

**Review by abnegate — APPROVED** [review]

**Pull request merged**